### DISCUSSION TITLE
DNS check unsuccessful, yet the domain resolves to github pages

### DISCUSSION THREAD

**Discussion by tjex** [community/community#75668] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Another person here struggling with DNS issues. 

Here you can see the error.
![Screenshot 2023-11-15 at 22 02 52](https://github.com/community/community/assets/83029642/0f6563ea-5d71-464e-ae71-8535edae8d8d)

It's complaining that the domain does not resolve to Github pages server, but it actually does.

![Screenshot 2023-11-15 at 22 17 23](https://github.com/community/community/assets/83029642/96871c71-c901-40e6-8ac8-9f7b068d4184)

output of `dig`.
i.e. Looking exactly as they should...

``` text
; <<>> DiG 9.10.6 <<>> tjex.net +nostats +nocomments +nocmd
;; global options: +cmd
;tjex.net.                      IN      A
tjex.net.               1383    IN      A       185.199.108.153
tjex.net.               1383    IN      A       185.199.111.153
tjex.net.               1383    IN      A       185.199.110.153
tjex.net.               1383    IN      A       185.199.109.153
```

```text
; <<>> DiG 9.10.6 <<>> www.tjex.net +nostats +nocomments +nocmd
;; global options: +cmd
;www.tjex.net.                  IN      A
www.tjex.net.           1062    IN      CNAME   tjex.github.io.
tjex.github.io.         1062    IN      A       185.199.111.153
tjex.github.io.         1062    IN      A       185.199.108.153
tjex.github.io.         1062    IN      A       185.199.110.153
tjex.github.io.         1062    IN      A       185.199.109.153
```

What's even stranger is that I tried for the hell of it to set the custom domain to the `www` subdomain, which gave me the same DNS error, but actually generated a certificate and resolved to the live version of my website... 

Makes no sense. Particularly considering these are my DNS records:

![Screenshot 2023-11-15 at 22 19 34](https://github.com/community/community/assets/83029642/d39b5bdd-492c-48b7-916b-36fac7c8041d)

Would be really great if someone from Github could chime in. Seen many occurrences of a staff member tinkering somewhere in the backend of Github pages and resolving this very quickly... Would be ace. 

Otherwise, I'm in a bind. Spoke to my DNS providers and on their end, everything looks fine. 

it's been over 24 hours since any DNS changes.

**Comment by tjex** [comment]
The page is served now, but without certificate. Still getting "DNS check unsuccessful", which is leading to no certificate being issued.

**Comment by peterroth** [comment]
Just to be sure, did you add a DNS TXT record on your domain hosting service? That validates that the hostname is yours, this is what the custom domain section checks. You can read about it on the [Verifying a domain for your user site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/verifying-your-custom-domain-for-github-pages#verifying-a-domain-for-your-user-site) page.

**Comment by tjex** [comment]
Thanks for the input Peter. 

I'd seen this step, but considered it to be from outdated documentation, as 
a) this button doesn't even exist in Github pages settings.
b) because the section in the sidebar is no longer titled "Code, automation and planning" (as referenced on the docs page), but just "Code and automation".
 
![Screenshot 2023-11-18 at 11 03 49](https://github.com/community/community/assets/83029642/e0e43bc3-51c1-4ddb-bbc4-1bd1e1675974)

**Reply by peterroth** [reply]
The documentation mentions you should click on your own profile photo and choose Settings, there you'll find the `Code, planning, and automation` section, below that there's the `Pages`. After clicking on that, you can see the `Add domain` button in the top right corner.
Or you can simply click here and follow the guide I linked above: https://github.com/settings/pages
I hope it helps.

**Reply by peterroth** [reply]
I checked the initial issue again and this site might carries the solution: https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site
Take a look at point 5. Based on the error message that's what probably missing in your case.

**Reply by tjex** [reply]
point 5 under which subheading? 
Thing is, I'm quite confident I've ticked off the requirements in that doc. Been over it a dozen times.

**Reply by peterroth** [reply]
[Configuring a subdomain](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain)

**Reply by tjex** [reply]
ah I see. however further down the page, there is a section for setting up an apex domain *and* a www subdomain, which requires the A records and CNAME record (pointing www to the github.io page).

https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain-and-the-www-subdomain-variant

Or have I misunderstood something?..

**Reply by tjex** [reply]
or maybe you're meaning that I'm missing the CNAME altogether? In which case it's shown at the bottom of my DNS settings (screenshot in the original post)

**Reply by peterroth** [reply]
I just checked your page's source (tjex.net repository) and currently there is no CNAME file.
Plus, on the picture [where you detail what DNS records you have at your DNS provider's page](https://user-images.githubusercontent.com/83029642/283250321-d39b5bdd-492c-48b7-916b-36fac7c8041d.png) the CNAME points to tjex.github.io, however that doesn't exist, the repository you created is tjex.net, it surely leads to problems.

**Reply by tjex** [reply]
yes! That's it! Thanks so much @peterroth 🙏

**Comment by tjex** [comment]
Also, since the original post, my site was accessible at tjex.net (with https being crossed out). 

![Screenshot 2023-11-18 at 11 11 57](https://github.com/community/community/assets/83029642/7339e88f-a0d9-498f-bcfa-b0b3aca9aa76)

But now after a few days I removed the domain from GitHub pages settings, and tried adding it again. Now tjex.net shows the GitHub Pages 404 again... 🤷‍♂️

(It wasn't a cache issue. I'd cleared my cache many times along the way)

I've also since set a `CAA` for letsencrypt.org over 72hours ago now (despite my registrar reportedly offering this by default). Didn't help.

**Comment by tjex** [comment, marked as answer]
...so. In the end, it was because my repository name did not match the `<user>.github.io` naming scheme 🤦‍♂️

Thanks Peter for pointing it out [here](https://github.com/orgs/community/discussions/75668#discussioncomment-7607274)

[Explained here in the docs under step 3](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site#creating-a-repository-for-your-site)

Frustrating as I already had this repo from when hosting my website elsewhere, so I never looked at the documentation for creating the repo (i.e. from the start...).

There is also the case of the CNAME file within the repo.

[Here under step 4](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain), the docs refer to the CNAME file needing to be present when building from a branch. 

It then states:

>  If you are publishing your site with a custom GitHub Actions workflow, no CNAME file is created. 

This is not super clear whether a CNAME file is needed when building from a workflow (or whether the GitHub workflow creates it itself when building the site). My assumption is that the user still needs to add it manually, I've also seen another repo with a CNAME file in the `/public` folder. 

Thing is that my site is now functioning and resolving correctly *without* the CNAME file. But it could be a case of errors down the track / unstable behaviour... So I'm going to add one anyway just in case.

Also [here](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site), there is the comment:

> A CNAME file in your repository file does not automatically add or remove a custom domain. Instead, you must configure the custom
> domain through your repository settings or through the API.

All in all, the documentation isn't super clear about whether a CNAME file is needed when building by workflow, but I suppose it can't hurt to include one.

**Reply by carlescufi** [reply]
As of 2024, this answer is not really correct. It is perfectly possible to  have a project website (i.e. one whose repo is not called <username>.github.io) using a custom domain with the DNS checks passing. @tjex are you sure you let 24h pass between the moment you changed your DNS settings in the Hosting provider and the moment you ran the DNS check on the GitHub repo?

**Reply by tjex** [reply]
well, I can't be sure just from memory as it was back last year, but throughout this process the only change to the respository name I made was from the original to tjex.github.io.

I wrote this:

> I've also since set a CAA for letsencrypt.org over 72hours ago now (despite my registrar reportedly offering this by default). Didn't help.

Which would then mean that yes, 24 hours must have passed as this CAA attempt to fix was made after the initial setup didn't work.

Also, as pointed out, the official documentation does state the naming requirement for the repository name as `<user>.github.io`. I see this consistently enough around GitHub too. The repositories I see that don't follow this convention are hosting their websites elsewhere. 

What would be best for your claim is if you have an example of a GitHub pages repository that doesn't follow this naming convention. Then it would be proven.

All I can say is that immediately after changing my repository name to the scheme stipualted in the documentation, the DNS resolved correctly.

**Reply by carlescufi** [reply]
> Also, as pointed out, the official documentation does state the naming requirement for the repository name as <user>.github.io. I see this consistently enough around GitHub too. The repositories I see that don't follow this convention are hosting their websites elsewhere.

It states that for **user sites** the repo name must be <user>.github.io. This is not the case for **project sites**:
https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#types-of-github-pages-sites

> What would be best for your claim is if you have an example of a GitHub pages repository that doesn't follow this naming convention. Then it would be proven.

There are many out there, let me point you to one:
https://stackoverflow.com/questions/72488329/github-pages-project-site-is-at-the-domain-root-when-configuring-custom-domain

**Comment by tjex** [comment]
I am building a user site. That's what this issue is about, sorry if that wasn't 
clear.

Well, the website posted in the stackoverflow thread (https://noymer.tk/) 
returns the exact same error I was having.

Perhaps the page is down anyway since then, perhaps since then the setup 
requirements have changed on GitHub's side since the stackoverflow answer (June 
2022), in any case, the page doesn't resolve, so it's not a good example unfortunately.
